arcgis-ios-sdk-samples/Scenes/Calculate 3D distance/Calculate3DDistanceViewController.swift

```diff
@@ -0,0 +1,283 @@
+        let redPoint = redGraphic.geometry as! AGSPoint
+        let greenPoint = greenGraphic.geometry as! AGSPoint
+        
+        let spherical1 = simd_double3(redPoint.x, redPoint.y, redPoint.z)
+        let spherical2 = simd_double3(greenPoint.x, greenPoint.y, greenPoint.z)
+        
+        let cartesian1 = cartesian(fromSpherical: spherical1)
+        let cartesian2 = cartesian(fromSpherical: spherical2)
+        
+        let distance = Measurement(
+            value: simd_distance(cartesian1, cartesian2),
+            unit: UnitLength.meters
+        )
+        distanceLabel.text = String(format: "Distance: %@", measurementFormatter.string(from: distance))
+    }
+    
+    /// Converts a spherical point to a cartesian point.
+    ///
+    /// - Parameter spherical: A point in the spherical coordinate system.
+    /// - Returns: A point in the cartesian coordinate system.
+    func cartesian(fromSpherical spherical: simd_double3) -> simd_double3 {
+        let earthRadius = Measurement(value: 6_371_000, unit: UnitLength.meters)
+        
+        let latitude = Measurement<UnitAngle>(value: spherical.x, unit: .degrees)
+            .converted(to: .radians)
+        let longitude = Measurement<UnitAngle>(value: spherical.y, unit: .degrees)
+            .converted(to: .radians)
+        let radius = earthRadius + Measurement(value: spherical.z, unit: .meters)
+        return simd_double3(
+            earthRadius.value * cos(latitude.value) * cos(longitude.value),
+            earthRadius.value * cos(latitude.value) * sin(longitude.value),
+            radius.value * sin(latitude.value)
+        )
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["Calculate3DDistanceViewController"]
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        drawStatusObservation = sceneView.observe(\.drawStatus, options: .initial) { [weak self] (sceneView, _) in
+            DispatchQueue.main.async {
+                guard sceneView.drawStatus == .completed,
+                    let self = self else {
+                        return
+                }
+                // Start the animation.
+                self.startAnimatingGraphics()
+                self.drawStatusObservation = nil
+            }
+        }
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        stopAnimatingGraphics()
+    }
+}
+
+/// A point in a two-dimensional coordinate system.
+struct Point: Equatable {
+    /// The x-coordinate of the point.
+    var x: Double
+    /// The y-coordinate of the point.
+    var y: Double
+}
+
+/// A function of a single variable.
+protocol Function {
```

Comment:
Instead of just `Function` what about `PositionFunction` to make it a bit clearer what the function does.

Reply:
Interesting suggestion. While the absolute value function is being used to calculate a position in this sample, a `Function` type isn't sample specific. It is possible that other samples could use `AbsoluteValueFunction` or declare their own function type to calculate something other than a position.

It might actually make more sense to move the `Function` code out of the sample since it isn't ArcGIS related. That would simplify the sample view controller while emphasizing that these `Function` components are reusable. Then if `PositionFunction` would be clearer, it can be declared as a private typedef to `AbsoluteValueFunction` inside the sample view controller source file.

Reply:
Moving things into other files and adding reusable components would go against some design ideas we've [discussed before](https://github.com/Esri/arcgis-runtime-samples-ios/pull/464).

As far as names, are you happy with calling `apply(to x: ...)` to get the next position?

Reply:
My understanding is that it is not desirable to make `ArcGIS`-related components reusable. We do have `ArcGIS`-unrelated reusable components in the sample app, such as [the methods for presenting an alert](https://github.com/Esri/arcgis-runtime-samples-ios/pull/516). Is my understanding incorrect? If so, please correct me.

I do like the name `apply(to:)` since it mirrors what a [mathematical function](https://en.wikipedia.org/wiki/Function_(mathematics)) actually does. We could rename it to `position(for:)`, but that seems a little redundant, assuming the function is named accordingly. My thought is that the names `redPositionFunction` and `greenPositionFunction` imply that the output is a position. What do you think?

Reply:
@dg0yal, I'll defer to you.  Do the names look okay?